cmd/sponged/handlers/data.go

```diff
@@ -0,0 +1,49 @@
+// Package handlers contains the handler logic for processing requests.
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ardanlabs/kit/db"
+	"github.com/ardanlabs/kit/web/app"
+	"github.com/coralproject/shelf/internal/sponge/item"
+)
+
+const defaultVersion = 1
+
+// dataHandle maintains the set of handlers for the data api.
+type dataHandle struct{}
+
+// Data fronts the access to the data service functionality.
+var Data dataHandle
+
+//==============================================================================
+
+// Upsert receives POSTed data, itemizes it then Upserts it via the item service
+// 204 SuccessNoContent, 400 Bad Request, 404 Not Found, 500 Internal
+func (dataHandle) Upsert(c *app.Context) error {
+
```

Comment:
Don't need this space if there is no comment. But why not put comments on all these separated code blocks. I have no idea what InferIDFromData does nor why it is called when it is called.